### ISSUE TITLE
Add the ability to quickly see orders that used discount by making 'use count' clickable in the Discounts list table

### ISSUE THREAD

**Issue by nutsandbolts** [awesomemotive/easy-digital-downloads#7361]
### Explain your enhancement (please be detailed)
A customer has requested the ability to sort/view purchases based on the discount code used, either by searching for the code in Payment History or by clicking the number of uses in the list of discount codes.

### Justification or use case
This would be useful to (1) quickly see which customers/purchases used the discount code, (2) determine whether a particular code was more effective than others, and/or (3) find an order you know was discounted (but may not have customer email or order number handy) to resolve an issue for a customer.

**Note:** Pippin mentioned the following, though I wasn't able to get it to work locally: 
```
Core already supports searching orders by discount code. Prefix search with
“discount: {code here}”
```
Possible user error on my part though I tried it a couple of different ways and verified the code existed and had been used.

**Comment by FPCSJames** [comment]
This search method does seem to be working for me. However, it would still be helpful to have a discount code column in Payment History, among other noted items above.

**Comment by cklosowski** [comment]
It seems to be working for me too:
![AwesomeScreenshot-Payments-EDD-DEV-WordPress-2019-07-23-10-07-91](https://user-images.githubusercontent.com/1390124/61731623-bd735b00-ad30-11e9-8573-99b426bdfb21.png)

@fpcorso The problem with a 'Discount Column' is that this can be an easily crowded area of the admin screen. With EDD core, it's pretty minimal, but as extensions add columns it quickly gets out of control.

We'll keep that suggestion in mind for the future.

**Comment by cklosowski** [comment]
I'm updating the title of this to provide an easier method of seeing the orders that used a discount by making the use count clickable in the discounts list table.

**Comment by FPCSJames** [comment]
One definite deficiency with the search method: there's no accurate count. As in your screenshot, it shows 0s for all the statuses, and no total on the right side of the table (or anywhere else).

**Comment by cklosowski** [comment]
@FPCSJames,

That should be resolved with 3.0, since the query classes are far more powerful in searching for related content, instead of through serialized arrays.

**Comment by cklosowski** [comment]
In EDD 3.0 we've fixed this by adding a 'View Orders' action link to each download, to filter the orders list table to orders that have used the discount.
